code/game/mecha/mecha.dm

```diff
@@ -1021,41 +1021,34 @@ var/year_integer = text2num(year) // = 2013???
 
 //////////////////////////////////////// Action Buttons ///////////////////////////////////////////////
 
-/obj/mecha/proc/GrantActions(var/mob/living/user, var/human_occupant = 0)
+/obj/mecha/proc/GrantActions(mob/living/user, human_occupant = 0)
 	if(human_occupant)
-		eject_action.chassis = src
-		eject_action.Grant(user)
+		eject_action.Grant(user, src)
+	internals_action.Grant(user, src)
+	cycle_action.Grant(user, src)
+	lights_action.Grant(user, src)
+	stats_action.Grant(user, src)
+	strafing_action.Grant(user, src)
 
-	internals_action.chassis = src
-	internals_action.Grant(user)
 
-	cycle_action.chassis = src
-	cycle_action.Grant(user)
-
-	lights_action.chassis = src
-	lights_action.Grant(user)
-
-	stats_action.chassis = src
-	stats_action.Grant(user)
-
-	strafing.chassis = src
-	strafing.Grant(user)

```

Comment:
removed some lines by giving the chassis assignment code to action/innate/mecha/Grant()